port/unix/omrintrospect.c

```diff
@@ -389,8 +389,8 @@ barrier_update_r(barrier_r *barrier, int new_value)
 		old_value = barrier->in_count;
 	} while (compareAndSwapUDATA((uintptr_t *)&barrier->in_count, old_value, old_value + difference) != old_value);
 
-	if (old_value < 0 || (old_value == 0 && barrier->initial_value != 0)) {
-		int restore_value = old_value;
+	if (old_value == 0 && barrier->initial_value != 0) {
+		uintptr_t restore_value = old_value;
 
```

Comment:
This value is being atomically swapped into `barrier->in_count`. What is the type of `barrier->in_count`?

Reply:
It's a `volatile uintptr_t`. The complete `barrier_r` type is as follows:
```c++
typedef struct {
	int descriptor_pair[2];
	volatile uintptr_t in_count;
	volatile uintptr_t out_count;
	uintptr_t initial_value;
	volatile uintptr_t released;
} barrier_r;
```